### ISSUE TITLE
Allow Int#chr to accept an optional encoding, much like String.new(bytes : Bytes, encoding : String)

### ISSUE THREAD

**Issue by postmodern** [crystal-lang/crystal#11258]
## Discussion

Allow `Int#chr` to accept an optional encoding for converting a byte value to the encodings defined character for said byte value. [String.new(bytes,encoding)](https://crystal-lang.org/api/1.1.1/String.html#new(bytes:Bytes,encoding:String,invalid:Symbol?=nil):String-class-method) already allows specifying a custom encoding, so I feel like `Int#chr` should as well.

**Comment by straight-shoota** [comment]
The idea does not sound very convincing. It seems like a very rare use case. `String.new` should be reasonably easy to use. Or you can implement a custom single character encoding method if you need to. But I don't see this as a common problem.

Can you explain why you think this would be generally useful as a standard lib feature?

**Comment by HertzDevil** [comment]
Ths two methods are not related; `String.new` relates to UTF-8 _byte sequences_, whereas `Int#chr` deals with UTF-8 _codepoints_. Not all encodings define their characters in terms of codepoints, so such a generalization isn't possible here.

**Comment by asterite** [comment]
Right. It seems you can pass an encoding to Ruby's chr, but in my mind that doesn't make much sense. Maybe it only works for one byte, but more bytes aren't supported? But I don't see that as very useful.

Maybe a method Char.new(byte, encoding) and Char.new(bytes, encoding) would make sense, where a single char is decoded from a byte or a byte sequence?
